custom_components/goldair_climate/device.py

```diff
@@ -100,9 +101,16 @@ async def async_inferred_type(self):
 
     async def async_refresh(self):
         last_updated = self._get_cached_state()["updated_at"]
-        if self._refresh_task is None or time() - last_updated >= self._CACHE_TIMEOUT:
-            self._cached_state["updated_at"] = time()
-            self._refresh_task = self._hass.async_add_executor_job(self.refresh)
+        try:
+            self._lock.acquire()
+            if (
+                self._refresh_task is None
+                or time() - last_updated >= self._CACHE_TIMEOUT
+            ):
+                self._cached_state["updated_at"] = time()
+                self._refresh_task = self._hass.async_add_executor_job(self.refresh)
+        finally:
+            self._lock.release()
```

Comment:
I can see the benefit of this improvement and planned to come back and re-work it at some point.

I'm trying to keep this as asynchronous as possible. It may be my lack of experience with Python, but my implementation seemed to be an asynchronous debounce - if there's already a refresh task, simply await the results without making the call again. Or is that actually running the task a second time? Possibly the specs need to be updated to detect extra calls first.